hardware_interface/test/fake_components/test_generic_system.cpp

```diff
@@ -0,0 +1,698 @@
+// Copyright (c) 2021 PickNik, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Denis Stogl
+
+#include <gmock/gmock.h>
+#include <string>
+#include <unordered_map>
+
+#include "hardware_interface/loaned_command_interface.hpp"
+#include "hardware_interface/loaned_state_interface.hpp"
+#include "hardware_interface/resource_manager.hpp"
+#include "ros2_control_test_assets/descriptions.hpp"
+#include "ros2_control_test_assets/components_urdfs.hpp"
+
+class TestGenericSystem : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    hardware_robot_2dof_ =
```

Comment:
couldn't this be replaced with something from test assets? should we add this there instead?

Reply:
I could add this there instead. But unfortunately can not be replaced because of plugin-used. 

Still, I would vote to keep this here since it is a very specific URDF for the FakeSystem implementation.

Reply:
:+1: